hal/src/main/native/include/HAL/DriverStation.h

```diff
@@ -79,9 +86,19 @@ struct HAL_JoystickDescriptor {
   uint8_t povCount;
 };
 
+struct HAL_MatchInfo {
+  char* eventName;
+  HAL_MatchType matchType;
+  uint16_t matchNumber;
+  uint8_t replayNumber;
+  uint8_t* gameSpecificMessage;
```

Comment:
uint8_t not char?

Reply:
I think it's safe to assume there's no embedded NUL characters, so it can be a simple nul-terminated string, and you can remove the size member of the structure.

Reply:
(this will also make it simply a string at the Java level too)

Reply:
The netcomm API has a char * with no length for name, and the uint8_t ptr with length for game specific. This makes it seem like game specific is not guaranteed to be null terminated, or even just a string of text. And even at the Hal to wpilib API level, we use char* for things definitely strings, and uint8_ts for stuff not for sure length.

Reply:
I'm pretty sure that's an artifact of the way that netcomm is getting the data (at the protocol level it's structured so the game-specific data completely fills the tag, so I think netcomm is just copying the entire tag out and keeping the types).  Keep in mind that the DS has a textbox for entering the game-specific data, so people are going to be hard-pressed to enter embedded nul characters or other binary data.

Reply:
Ok I fixed both off these.